Cubical/ZCohomology/RingStructure/GradedCommutativity.agda

```diff
@@ -0,0 +1,1082 @@
+                     (subst coHomK (+'-comm 1 (suc (suc n)))
+                      (_⌣ₖ_ {n = suc zero} {m = suc (suc n)} ∣ loop j ∣ₕ
+                        ∣ compPath-filler (merid a) (sym (merid (ptSn (suc n)))) (~ r) i ∣ₕ))})
+        (hcomp (λ r →
+          λ { (i = i0) → (-ₖ-gen (suc (suc n)) 1 p q ∘
+                            (subst coHomK (+'-comm 1 (suc (suc n)))))
+                              ∣ rCancel (merid (ptSn (suc (suc n)))) (~ k ∧ r) j ∣
+            ; (i = i1) → (-ₖ-gen (suc (suc n)) 1 p q ∘
+                            (subst coHomK (+'-comm 1 (suc (suc n)))))
+                              ∣ rCancel (merid (ptSn (suc (suc n)))) (~ k ∧ r) j ∣
+            ; (k = i0) → help₂ r i j
+            ; (k = i1) → -ₖ-gen (suc (suc n)) 1 p q
+                           (subst coHomK (+'-comm 1 (suc (suc n)))
+                            (_⌣ₖ_ {n = suc zero} {m = suc (suc n)} ∣ loop j ∣ₕ
+                              (Kn→ΩKn+1 _ ∣ a ∣ₕ i)))})
+   (-ₖ-gen (suc (suc n)) 1 p q
+                           (subst coHomK (+'-comm 1 (suc (suc n)))
+                            (_⌣ₖ_ {n = suc zero} {m = suc (suc n)} ∣ loop j ∣ₕ
+                              (Kn→ΩKn+1 _ ∣ a ∣ₕ i)))))
+    where
+    P : Path _ (Kn→ΩKn+1 (suc (suc (n + 0))) (0ₖ _))
+               (Kn→ΩKn+1 (suc (suc (n + 0))) (_⌣ₖ_ {n = (suc n)} {m = suc zero} ∣ a ∣ ∣ base ∣))
+    P i = Kn→ΩKn+1 (suc (suc (n + 0)))
+            ((sym (gradedComm-elimCase-left n (evenOrOdd (suc n)) (inr tt) a base
+              ∙ cong (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt)) (transp0₁ n)) i))
+
+    help₁ : (P ∙∙ ((λ i j → _⌣ₖ_ {n = suc (suc n)} {m = suc zero} ∣ merid a j ∣ₕ ∣ loop i ∣ₕ)) ∙∙ sym P)
+         ≡ ((λ i → Kn→ΩKn+1 _ (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt)
+                                   (transp0₁ n (~ i))))
+        ∙∙ (λ i j → Kn→ΩKn+1 _ (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt)
+                                  (subst coHomK (+'-comm 1 (suc n)) (∣ loop i ∣ₕ ⌣ₖ ∣ a ∣ₕ))) j)
+        ∙∙ (λ i → Kn→ΩKn+1 _ (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt)
+                                   (transp0₁ n i))))
+    help₁ k i j =
+        ((λ i → (Kn→ΩKn+1 (suc (suc (n + 0))))
+                  (compPath-filler'
+                    ((gradedComm-elimCase-left n (evenOrOdd (suc n)) (inr tt) a base))
+                   (cong (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt))
+                    (transp0₁ n)) (~ k) (~ i)))
+       ∙∙ (λ i j → (Kn→ΩKn+1 _
+                     (gradedComm-elimCase-left n (evenOrOdd (suc n)) (inr tt) a (loop i) k) j))
+       ∙∙ λ i → (Kn→ΩKn+1 (suc (suc (n + 0))))
+                   (compPath-filler'
+                     ((gradedComm-elimCase-left n (evenOrOdd (suc n)) (inr tt) a base))
+                    (cong (-ₖ-gen (suc n) 1 (evenOrOdd (suc n)) (inr tt))
+                     (transp0₁ n)) (~ k) i)) i j
+
+    help₂ : I → I → I → coHomK _
+    help₂ r i j =
+      hcomp (λ k →
```

Comment:
Yikes, I think we might have a new high dimension hcomp master! @ecavallo

Reply:
Watch out, he's going to turn it into one line